docs/querying/datasource.md

```diff
@@ -334,7 +336,7 @@ Native join datasources have the following properties. All are required.
 |`right`|Right-hand datasource. Must be of type `lookup`, `query`, or `inline`. Note that this is more rigid than what Druid SQL requires.|
 |`rightPrefix`|String prefix that will be applied to all columns from the right-hand datasource, to prevent them from colliding with columns from the left-hand datasource. Can be any string, so long as it is nonempty and is not be a prefix of the string `__time`. Any columns from the left-hand side that start with your `rightPrefix` will be shadowed. It is up to you to provide a prefix that will not shadow any important columns from the left side.|
 |`condition`|[Expression](../misc/math-expr.md) that must be an equality where one side is an expression of the left-hand side, and the other side is a simple column reference to the right-hand side. Note that this is more rigid than what Druid SQL requires: here, the right-hand reference must be a simple column reference; in SQL it can be an expression.|
-|`joinType`|`INNER` or `LEFT`.|
+|`joinType`|`INNER`, `LEFT`|
 
```

Comment:
Add 'RIGHT' ?

But I notice in the descriptive paragraph you are using 'left' and right' to indicate the driving and joining tables ... unfortunately that might all need to change to accommodate the LEFT and RIGHT JOIN modes.   

Sorry for opening up a rats nest here ...

Reply:
In Druid,`LEFT` is a more generic term for `OUTER`. `OUTER` meant "LEFT OUTER" in older versions of SQL. So, the term "RIGHT OUTER" had to be introduced. Given that, it seems modern syntax omits "OUTER" to give just `LEFT` and `RIGHT`. Reference: [left join](https://www.w3schools.com/sql/sql_join_left.asp), [right join](https://www.w3schools.com/sql/sql_join_right.asp).

However, [this reference](https://mode.com/sql-tutorial/sql-outer-joins/) says, "FULL OUTER JOIN is also refered to as OUTER JOIN." which didn't used to be true on many databases.

Checking the code (`Parser.jj`) finds that, in the Calcite dialect, `OUTER` is synonymous with `LEFT`.